File: lista03W/zad1/Pieciokat.java
public class Pieciokat extends Figura {
  private double bok;
  
  public Pieciokat(double bok) {
    this.bok = bok;
  }
  
  public double getPole() {
    return (5 * bok * bok) / (4 * Math.tan(Math.PI / 5));
  }
  
  public double getObwod() {
    return 5 * bok;
  }
  
  public String getNazwa() {
    return "Pieciokat foremny";
  }
}

File: lista03W/zad2/pieciokat.cpp
#include "pieciokat.hpp"

Pieciokat::Pieciokat(double b) {
  if(b <= 0)
    throw std::invalid_argument("Bok pieciokata musi byc dodatni.");

  bok = b;
}

double Pieciokat::getPole() const {
  return (5 * bok * bok) / (4 * tan(M_PI / 5));
}

double Pieciokat::getObwod() const {
  return 5 * bok;
}

std::string Pieciokat::getNazwa() const {
  return "Pieciokat foremny";
}